EDEngineer.Models/Operations/MaterialTradeOperation.cs

```diff
@@ -4,22 +4,30 @@ namespace EDEngineer.Models.Operations
 {
     public class MaterialTradeOperation : JournalOperation
     {
-        public string IngredientRemoved { get; set; }
-        public int RemovedQuantity { get; set; }
+        private Dictionary<string, int> _changes = new Dictionary<string, int>();
 
-        public string IngredientAdded { get; set; }
-        public int AddedQuantity { get; set; }
+        public void AddIngredient(string ingredient, int count)
+		{
+            if (!_changes.ContainsKey(ingredient))
+                _changes.Add(ingredient, 0);
+            _changes[ingredient] += count;
+        }
+
+        public void RemoveIngredient(string ingredient, int count)
+        {
+            if (!_changes.ContainsKey(ingredient))
+                _changes.Add(ingredient, 0);
```

Comment:
some formatting to do in this file too (brackets on all `if` + tabs to four spaces)

Reply:
Done